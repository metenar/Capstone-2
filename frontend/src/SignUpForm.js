import React from "react";
import {useHistory} from "react-router-dom";
import 'bootstrap/dist/css/bootstrap.css';
import { Form, FormGroup, Label, Input} from "reactstrap";
import {Button} from "reactstrap"
import { useFormik } from 'formik';

const validate = values => {
    const errors = {};
    if (!values.username) {
        errors.username = 'Required';
    } else if (values.username.length > 15) {
        errors.username = 'Must be 15 characters or less';
    }
  
    if (!values.password) {
        errors.password = 'Required';
    } else if (values.password.length < 6) {
        errors.password = 'Must be at least 6 characters';
    }

    if (!values.first_name) {
        errors.first_name = 'Required';
    } else if (values.first_name.length > 15) {
        errors.first_name = 'Must be 15 characters or less';
    }

    if (!values.last_name) {
        errors.last_name = 'Required';
    } else if (values.last_name.length > 20) {
        errors.last_name = 'Must be 20 characters or less';
    }
    
    if (!values.email) {
        errors.email = 'Required';
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
        errors.email = 'Invalid email address';
    }

    return errors;
  };

const SignUpForm=({signup})=>{
    const history= useHistory();
    const formik = useFormik({
        initialValues : {
          username : '',
          first_name: '',
          last_name: '',
          email: '',
          password : '',
          image: ''
        },
        validate,
        onSubmit      : async (values) => {
          try {
            let res=await signup(values);
            if(res.success){
              history.push('/')
          }
          } catch (err) {
            formik.errors.password = err;
          }
        }
      });
    
    return (
        <section className="col-md-4 m-auto">
        <Form onSubmit={formik.handleSubmit}>
            <h2>SignUp</h2>
            <FormGroup>
                <Label for="username">Username </Label>
                <Input 
                    type="text"
                    id="username"
                    name="username"
                    value={formik.values.username}
                    placeholder="Enter username"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.username && formik.errors.username ? <div className="alert alert-danger">{formik.errors.username}</div>:null}
            </FormGroup>
            <FormGroup>
                <Label for="password">Password </Label>
                <Input 
                    type="password"
                    id="password"
                    name="password"
                    value={formik.values.password}
                    placeholder="Enter your password"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.password && formik.errors.password ? <div className="alert alert-danger">{formik.errors.password}</div>:null}
            </FormGroup>
            <FormGroup>
                <Label for="first_name">First Name </Label>
                <Input 
                    type="text"
                    id="first_name"
                    name="first_name"
                    value={formik.values.first_name}
                    placeholder="Enter your First Name"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.first_name && formik.errors.first_name ? <div className="alert alert-danger">{formik.errors.first_name}</div>:null}
            </FormGroup>
            <FormGroup>
                <Label for="last_name">Last Name </Label>
                <Input 
                    type="text"
                    id="last_name"
                    name="last_name"
                    value={formik.values.last_name}
                    placeholder="Enter your last Name"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.last_name && formik.errors.last_name ? <div className="alert alert-danger">{formik.errors.last_name}</div>:null}
            </FormGroup>
            <FormGroup>
                <Label for="email">Email </Label>
                <Input 
                    type="email"
                    id="email"
                    name="email"
                    value={formik.values.email}
                    placeholder="Enter your email"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.email && formik.errors.email ? <div className="alert alert-danger">{formik.errors.email}</div>:null}
            </FormGroup>
            <FormGroup>
                <Label for="image">Avatar image </Label>
                <Input 
                    type="text"
                    id="image"
                    name="image"
                    value={formik.values.image}
                    placeholder="Enter your image"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.image && formik.errors.image ? <div className="alert alert-danger">{formik.errors.image}</div>:null}
            </FormGroup>
            <Button type="submit" color="primary">Submit</Button>
        </Form>
        </section>
    )
}
export default SignUpForm;