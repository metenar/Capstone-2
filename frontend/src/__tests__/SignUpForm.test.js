import { render,screen} from '@testing-library/react';
import { MemoryRouter, Route, Switch} from "react-router-dom";
import SignUpForm from '../SignUpForm';
import '@testing-library/jest-dom'
import CurrentUserContext from "../CurrentUserContext"

function renderSignUp(currentUser) {

    return render(
        <MemoryRouter initialEntries={["/signup"]}>
            <CurrentUserContext.Provider value={currentUser}>
                <Switch>
                    <Route exect path="/signup">
                        <SignUpForm />
                    </Route>
                </Switch>
            </CurrentUserContext.Provider>
        </MemoryRouter>
    );
  }

  const currentUser=[
    {
      username:"metenar",
      firstName:"Mete",
      lastName:"NAR",
      email:'metenar@gmail.com',
      password:"",
      image:"www.mete.jpg"
    }]

test('renders without crushing', () => {
  renderSignUp({currentUser});
});

test('should match the snapshot', () => {
  const {asFragment}=renderSignUp({currentUser});
  expect(asFragment()).toMatchSnapshot()
});

test('should First Name visible', ()=>{
    renderSignUp({currentUser});
    expect(screen.queryAllByText('First Name')).not.toBeNull();
})