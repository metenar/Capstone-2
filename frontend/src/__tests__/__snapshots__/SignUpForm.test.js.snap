// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4 m-auto"
  >
    <form
      class=""
    >
      <h2>
        SignUp
      </h2>
      <div
        class="form-group"
      >
        <label
          class=""
          for="username"
        >
          Username 
        </label>
        <input
          class="form-control"
          id="username"
          name="username"
          placeholder="Enter username"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="password"
        >
          Password 
        </label>
        <input
          class="form-control"
          id="password"
          name="password"
          placeholder="Enter your password"
          type="password"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="first_name"
        >
          First Name 
        </label>
        <input
          class="form-control"
          id="first_name"
          name="first_name"
          placeholder="Enter your First Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="last_name"
        >
          Last Name 
        </label>
        <input
          class="form-control"
          id="last_name"
          name="last_name"
          placeholder="Enter your last Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="email"
        >
          Email 
        </label>
        <input
          class="form-control"
          id="email"
          name="email"
          placeholder="Enter your email"
          type="email"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="image"
        >
          Avatar image 
        </label>
        <input
          class="form-control"
          id="image"
          name="image"
          placeholder="Enter your image"
          type="text"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
